src/loader/mod.rs

```diff
@@ -211,20 +215,21 @@ pub fn load_cmdline<M: GuestMemory>(
     guest_addr: GuestAddress,
     cmdline: &Cmdline,
 ) -> Result<()> {
-    let len = cmdline.as_str().len();
-    if len == 0 {
-        return Ok(());
-    }
+    let cmdline_cstring = cmdline
```

Comment:
Since we only use the bytes I think we can just keep those bytes around by directly calling here `as_bytes_with_null`. We should add a comment here as well that Linux expects the command line to end with a `\0` which is why we're doing this.